### PR TITLE
[CLOB-1054] add some telemetry for conditional order triggering

### PR DESCRIPTION
**Pull request by jayy04**
### Changelist
- add/update telemetry for conditional order triggering.

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/x/clob/keeper/untriggered_conditional_orders.go [hunk 1]
```diff
@@ -288,78 +291,132 @@ func (k Keeper) MaybeTriggerConditionalOrders(ctx sdk.Context) (triggeredConditi
...
 		// Set the modified untriggeredConditionalOrders back on the keeper field.
-		k.UntriggeredConditionalOrders[clobPairId] = untriggeredConditionalOrders
+		k.UntriggeredConditionalOrders[clobPairId] = untriggered
 	}
 
+	return allTriggeredOrderIds
+}
+
+// TriggerOrdersWithPrice triggers all untriggered conditional orders using the given price. It returns
+// a list of order ids that were triggered. This function is called in EndBlocker.
```
Comment (jonfung-dydx) [inline protocol/x/clob/keeper/untriggered_conditional_orders.go:324]:
```suggestion
// TriggerOrdersWithPrice triggers all untriggered conditional orders using the given price. It returns
// a list of order ids that were triggered. This function is called in EndBlocker.
// It removes all triggered conditional orders from the `UntriggeredConditionalOrders ` struct.
```

File: protocol/x/clob/keeper/untriggered_conditional_orders.go [hunk 2]
```diff
@@ -288,78 +291,132 @@ func (k Keeper) MaybeTriggerConditionalOrders(ctx sdk.Context) (triggeredConditi
...
 			indexerevents.StatefulOrderEventVersion,
 			indexer_manager.GetBytes(
 				indexerevents.NewConditionalOrderTriggeredEvent(
-					triggeredConditionalOrderId,
+					orderId,
 				),
 			),
 		)
+
+		telemetry.IncrCounterWithLabels(
```
Comment (jonfung-dydx) [inline protocol/x/clob/keeper/untriggered_conditional_orders.go:360]: pls use new metrics library!

Comment (jonfung-dydx) [reply protocol/x/clob/keeper/untriggered_conditional_orders.go:360]: also define new non-modular metric name.

JW why we moved this out of `MustTriggerConditionalOrder`?

File: protocol/x/clob/keeper/untriggered_conditional_orders.go [hunk 3]
```diff
@@ -288,78 +291,132 @@ func (k Keeper) MaybeTriggerConditionalOrders(ctx sdk.Context) (triggeredConditi
...
+) (triggeredOrderIds []types.OrderId) {
+	triggeredOrderIds = untriggered.PollTriggeredConditionalOrders(price)
+
+	// Emit metrics.
+	priceFloat, _ := price.Float32()
+	labels := []gometrics.Label{
+		metrics.GetLabelForStringValue(metrics.Type, priceType),
+		metrics.GetLabelForIntValue(metrics.PerpetualId, int(perpetualId)),
+	}
+	telemetry.SetGaugeWithLabels([]string{metrics.ClobConditionalOrderTriggerPrice}, priceFloat, labels)
```
Comment (jonfung-dydx) [inline protocol/x/clob/keeper/untriggered_conditional_orders.go:340]: new metrics library pls!

**Review by jonfung-dydx — APPROVED** [review]

**Comment by jonfung-dydx** [comment]
can you make a ticket for O(3N) -> O(N)?

File: protocol/x/clob/keeper/untriggered_conditional_orders.go (hunk 2 above)
Comment (jayy04) [reply protocol/x/clob/keeper/untriggered_conditional_orders.go:361]: so that we can add price type as a label (without passing it down as an argument)

**Pull request merged**